pkg/controller/clusterpool/clusterpool_controller_test.go

```diff
@@ -0,0 +1,206 @@
+package clusterpool
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/openshift/hive/pkg/apis"
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	"github.com/openshift/hive/pkg/apis/hive/v1/aws"
+	hivev1aws "github.com/openshift/hive/pkg/apis/hive/v1/aws"
+)
+
+const (
+	testNamespace     = "test-namespace"
+	hiveNamespace     = "hive"
+	testLeasePoolName = "aws-us-east-1"
+	credsSecretName   = "aws-creds"
+)
+
+func init() {
+	log.SetLevel(log.DebugLevel)
+}
+
+func TestReconcileClusterPool(t *testing.T) {
```

Comment:
Add test cases for the following.
1) Clusters that are not part of a pool do not count against pool size.
2) Clusters that have been claimed from the pool do not count against pool size.
3) Clusters that are part of a different pool do not count against pool size.
4) Clusters that are deleting do not count against pool size.